app/src/controllers/pagination/reducer.js

```diff
@@ -14,7 +14,24 @@
  * limitations under the License.
  */
 
-import { fetchReducer } from 'controllers/fetch';
+import { FETCH_SUCCESS, fetchReducer } from 'controllers/fetch';
+import { initialPaginationState } from './constants';
 
 export const paginationReducer = (namespace, initialState = {}) =>
   fetchReducer(namespace, { contentPath: 'page', initialState });
+
+export const alternativePaginationReducer = (state = initialPaginationState, { type, payload }) => {
```

Comment:
As we still need to manage `namespace`s inside this reducer, I suggest to extend the `fetchReducer` with some kind of callback called f.e. `payloadConverter` or smth similar, that will do necessary calculations before returning the value.
So the final `alternativePaginationReducer` should look like:
```javascript
export const alternativePaginationReducer = (namespace, initialState = {}) =>
  fetchReducer(namespace, { initialState }, payloadConverter);
```

Reply:
Done.